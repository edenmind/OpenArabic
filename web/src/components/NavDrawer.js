import * as React from 'react'

import { Box, Button, Divider, Drawer, List, ListItem } from '@mui/material'

import { Fragment } from 'react'
import { Link } from 'react-router-dom'
import PropTypes from 'prop-types'
import axios from 'axios'

function NavDrawer(props) {
  const [categories, setCategories] = React.useState([])

  React.useEffect(() => {
    axios
      .get(`${process.env.REACT_APP_API_URL}/categories`)
      .then((response) => {
        setCategories(response.data)
      })
      .catch((err) => console.log(err))
  }, [])

  return (
    <Drawer anchor={'left'} open={props.state.left} onClose={props.toggleDrawer('left', false)}>
      <Box sx={{ width: 250 }} role='presentation' onClick={props.toggleDrawer('left', false)} onKeyDown={props.toggleDrawer('left', false)}>
        <List>
          {categories.map((category, index) => (
            <ListItem key={index}>
              <Link to={`/texts/categories/${category.id}`}>
                <Button variant='text'>{category.name}</Button>
              </Link>
            </ListItem>
          ))}
        </List>

        {props.isAuthenticated && (
          <Fragment>
            <Divider>Administration</Divider>
            <List>
              {['Texts', 'Categories', 'Authors'].map((item, index) => (
                <ListItem key={index}>
                  <Link to={`/${item}`}>
                    <Button variant='text'>{item}</Button>
                  </Link>
                </ListItem>
              ))}
            </List>
          </Fragment>
        )}
      </Box>
    </Drawer>
  )
}

NavDrawer.propTypes = {
  isAuthenticated: PropTypes.bool,
  toggleDrawer: PropTypes.func.isRequired,
}

export default NavDrawer
