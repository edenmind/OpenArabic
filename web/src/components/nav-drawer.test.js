import NavDrawer from './nav-drawer.js'
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow' // ES6

const toggleDrawer = () => {}
it('renders without crashing', () => {
  const renderer = new ShallowRenderer()

  const isAuthenticated = false
  const state = {
    left: false
  }

  renderer.render(<NavDrawer isAuthenticated={isAuthenticated} toggleDrawer={toggleDrawer} state={state} />)
})

it('should match snapshot when authenticated and open', () => {
  const renderer = new ShallowRenderer()

  const isAuthenticated = true
  const state = {
    left: true
  }

  renderer.render(<NavDrawer isAuthenticated={isAuthenticated} toggleDrawer={toggleDrawer} state={state} />)

  const result = renderer.getRenderOutput()

  expect(result).toMatchSnapshot()
})
