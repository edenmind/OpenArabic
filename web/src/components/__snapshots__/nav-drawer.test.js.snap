// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when authenticated and open 1`] = `
<ForwardRef(Drawer)
  anchor="left"
  open={true}
>
  <ForwardRef(Box)
    role="presentation"
    sx={
      Object {
        "width": 250,
      }
    }
  >
    <ForwardRef(List) />
    <React.Fragment>
      <ForwardRef(Divider)>
        Administration
      </ForwardRef(Divider)>
      <ForwardRef(List)>
        <ForwardRef(ListItem)>
          <ForwardRef(Button)
            component={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/Texts"
          >
            Texts
          </ForwardRef(Button)>
        </ForwardRef(ListItem)>
        <ForwardRef(ListItem)>
          <ForwardRef(Button)
            component={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/Categories"
          >
            Categories
          </ForwardRef(Button)>
        </ForwardRef(ListItem)>
        <ForwardRef(ListItem)>
          <ForwardRef(Button)
            component={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/Authors"
          >
            Authors
          </ForwardRef(Button)>
        </ForwardRef(ListItem)>
        <ForwardRef(ListItem)>
          <ForwardRef(Button)
            component={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
            to="/Words"
          >
            Words
          </ForwardRef(Button)>
        </ForwardRef(ListItem)>
      </ForwardRef(List)>
    </React.Fragment>
  </ForwardRef(Box)>
</ForwardRef(Drawer)>
`;
